.github/workflows/pymeasure_CI.yml

```diff
@@ -10,16 +10,32 @@ jobs:
   docs_lint:
     name: Docs and Linting
     runs-on: "ubuntu-latest"
+    defaults:
+      run:
+        shell: bash -l {0}
     steps:
       - uses: actions/checkout@v2
-      - uses: actions/setup-python@v2
+      - name: Cache conda
+        uses: actions/cache@v2
+        env:
+          CACHE_NUMBER: 0  # Increase to reset the cache
+        with:
+          path: ~/conda_pkgs_dir
+          key: py3.8-ubuntu-latest-conda-${{ env.CACHE_NUMBER }}-${{ hashFiles('requirements/*.yml') }}
+      - name: Install pymeasure requirements
+        uses: conda-incubator/setup-miniconda@v2
         with:
           python-version: "3.8"
-      - name: Install docs dependencies
+          environment-file: requirements/pymeasure.yml
+          activate-environment: pymeasure
+          auto-update-conda: true
+          use-only-tar-bz2: true  # for caching to work properly
+      - name: Install dev requirements
+        run: conda env update -f requirements/pymeasure_dev.yml
+      - name: Conda info & list
         run: |
-          pip install wheel
-          pip install -r requirements/pymeasure_readthedocs.txt
```

Comment:
Good point -- we are using pip for RTD builds. Historically we used conda because it was difficult to pip install the Qt requirements. I wonder if that is still a limitation -- if not we could move away from conda. Otherwise, I think we should consider moving RTD to conda. If dependencies are missing, the doc pages can be dropped, so it is important to have all requirements in that job.

I like the mamba suggestion, and will look into it more.

Reply:
IDK how flexible RTD is regarding builds - never used it. If we can move that to conda, why not. I guess you can check a recent (or master-branch) CI log if there were any difficulties with pip & qt.

I would very much like to keep conda, I have virtually stopped using pip years ago, for scientific python + Windows, conda just makes so much sense.

Reply:
Yea, sticking with conda sounds good. I think mamba is something to keep an eye on and maybe try later this year.

Reply:
Mamba would be interesting, and I would be fine to switch. I meant conda as opposed to pip, mentally mamba counts as conda for me :D